ion_hash_test.go

```diff
@@ -0,0 +1,220 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package ionhash
+
+import (
+	"bytes"
+	"io/ioutil"
+	"strings"
+	"testing"
+
+	"github.com/amzn/ion-go/ion"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestIonHash(t *testing.T) {
+	parameters := ionHashDataSource(t)
+
+	for i := range parameters {
+		ionBinary := parameters[i].testCase
+		reader := ion.NewReaderBytes(ionBinary)
+
+		provider := parameters[i].provider
+		Traverse(t, reader, provider.getInstance())
+
+		if len(parameters[i].expectedHashLog.identityUpdateList) > 0 {
+			assert.Equal(t, provider.getUpdateHashLog(), parameters[i].expectedHashLog.identityUpdateList, parameters[i].hasherName+" failed")
```

Comment:
The expected value should go before the actual value. Switch these around.

Reply:
Updated